corehq/apps/app_manager/tests/test_practice_user_config.py

```diff
@@ -0,0 +1,129 @@
+from django.test import TestCase
+
+from corehq.apps.app_manager.exceptions import PracticeUserException
+from corehq.apps.app_manager.models import Application, BuildProfile
+from corehq.apps.domain.shortcuts import create_domain
+from corehq.apps.ota.utils import turn_on_demo_mode, turn_off_demo_mode
+from corehq.apps.users.models import CommCareUser
+
+from .app_factory import AppFactory
+from .util import TestXmlMixin
+
+
+class TestPracticeUserRestore(TestCase, TestXmlMixin):
+    """Tests for Practice Mobile Worker feature"""
+    file_path = ('data',)
+
+    @classmethod
+    def setUpClass(cls):
+        super(TestPracticeUserRestore, cls).setUpClass()
+        from corehq.apps.users.dbaccessors.all_commcare_users import delete_all_users
+        delete_all_users()
+        cls.domain = "practice-user-domain"
+
+        cls.factory = AppFactory(build_version='2.28.0', domain=cls.domain)
+        module, form = cls.factory.new_basic_module('register', 'case')
+        form.source = cls.get_xml('very_simple_form')
+
+        cls.project = create_domain(cls.domain)
+        cls.user = CommCareUser.create(cls.domain, 'test@main-domain.commcarehq.org', 'secret')
+
+    def assert_(self, app, expected_paths):
+        # test user-restore files created
+        # test resource blocks created
+        app.create_build_files
+
+    def tearDown(self):
+        turn_off_demo_mode(self.user)
+
+    @staticmethod
+    def _get_restore_resource(version, build_profile_id=None):
+        extra = "?profile={id}".format(id=build_profile_id) if build_profile_id else ""
+        return """
+        <partial>
+        <user-restore>
+           <resource id="practice-user-restore" version="{version}" descriptor="practice user restore">
+             <location authority="local">./practice_user_restore.xml</location>
+             <location authority="remote">./practice_user_restore.xml{extra}</location>
+           </resource>
+        </user-restore>
+        </partial>
+        """.format(version=version, extra=extra)
+
+    def test_app_specific(self):
+        turn_on_demo_mode(self.user, self.domain)
+        app = self.factory.app
+        app.practice_mobile_worker_id = self.user._id
+
+        # check suit contains user-restore resource
+        self.assertXmlPartialEqual(
+            self._get_restore_resource(self.user.demo_restore_id),
+            app.create_suite(),
+            "./user-restore"
+        )
+        # check 'files/practice_user_restore.xml' is included in the build files
+        app.create_build_files(save=True)
+        app.save()
+        self.assertTrue(app.lazy_fetch_attachment('files/practice_user_restore.xml'))
+
+    def test_profile_specific(self):
+        turn_on_demo_mode(self.user, self.domain)
+        app = self.factory.app
```

Comment:
These tests share the same app. It would be better to create a new app each time to have a clean slate.